vg/font.go

```diff
@@ -149,42 +159,73 @@ type FontExtents struct {
 
 // Extents returns the FontExtents for a font.
 func (f *Font) Extents() FontExtents {
-	bounds := f.font.Bounds(fixed.Int26_6(f.Font().FUnitsPerEm()))
-	scale := f.Size / Points(float64(f.Font().FUnitsPerEm()))
+	var (
+		// TODO(sbinet): re-use a Font-level sfnt.Buffer instead?
+		buf  sfnt.Buffer
+		ppem = fixed.Int26_6(f.font.UnitsPerEm())
+	)
+
+	met, err := f.font.Metrics(&buf, ppem, f.hinting)
+	if err != nil {
+		panic(fmt.Errorf("could not extract font extents: %v", err))
+	}
+	scale := f.Size / Points(float64(ppem))
 	return FontExtents{
-		Ascent:  Points(float64(bounds.Max.Y)) * scale,
-		Descent: Points(float64(bounds.Min.Y)) * scale,
-		Height:  Points(float64(bounds.Max.Y-bounds.Min.Y)) * scale,
+		Ascent:  Points(float64(met.Ascent)) * scale,
+		Descent: Points(float64(-met.Descent)) * scale, // sfnt-descent is >0
```

Comment:
Why do we change sign?

Reply:
we change the sign for backward compatibility with `vg.FontExtents` documentation.

I admit I pondered on changing the documentation and propagating the new definition of `Descent` (to follow the one of `x/image/font.Metrics`) but kept it for later.
this could be done in this PR.

Reply:
I think it's worth considering, but later.